### PR TITLE
Feature/sc 45590/prototype cloud functions partial deployment

### PR DESCRIPTION
**Pull request by pedrofsantoscom**
image

### CODE REVIEW

File: src/discoverer/discoverer.ts [hunk 1]
```diff
@@ -16,6 +17,9 @@ const getExports = (indexFilePath = "src/index.ts"): Record<string, string> => {
     const indexFile = project.getSourceFileOrThrow(indexFilePath);
 
     const exportSymbols = typeChecker.getExportsOfModule(indexFile.getSymbolOrThrow());
+    const firstExportedVariableStatement = exportSymbols.map((e) => e.getEscapedName());
+    logger.info(firstExportedVariableStatement);
+    kill();
```
Comment (pedrofsantoscom) [inline src/discoverer/discoverer.ts:22]: this file is not being used and its disabled. this code is just for testing so no problem leaving the kill() there.
im going back to this function later

File: src/index.ts [hunk 2]
```diff
@@ -57,21 +60,25 @@ async function main() {
     }
     const bundlerConfig: BundlerConfig = {
         ...bundlerConfigSpecResult.value,
-        concurrency,
+        // concurrency,
```
Comment (SirPhemmiey) [inline src/index.ts:63]: @pedrofsantoscom why is concurrency removed here? i thought it was required to speed up the process?

File: src/index.ts [hunk 3]
```diff
@@ -87,7 +94,7 @@ async function main() {
             return record;
         }, <Record<string, string>>{});
 
-    const diffResults = hashesDiffer(existingHashes ?? {}, newHashes);
+    const diffResults = hashesDiffer((forceDeploy ? {} : existingHashes) ?? {}, newHashes);
```
Comment (SirPhemmiey) [inline src/index.ts:97]: okay, this is where the forceDeploy you mentioned will deploy all functions, cool.

File: src/index.ts (hunk 2 above)
Comment (pedrofsantoscom) [reply src/index.ts:63]: doesnt work with the function i'm using. plus concurrency didn't made much difference even on my mac

**Comment by pedrofsantoscom** [comment]
https://app.shortcut.com/ajarhq/story/45590/prototype-cloud-functions-partial-deployment-script

**Comment by pedrofsantoscom** [comment]
[sc-45590]

**Comment by pedrofsantoscom** [comment]
https://app.shortcut.com/ajarhq/story/45590/prototype-cloud-functions-partial-deployment-script

**Pull request merged**